src/sign-in-xplat/MainPage.xaml.cs

```diff
@@ -48,36 +53,41 @@ private async void OnSignInClicked(object sender, EventArgs e)
                     .Build();
 
                 // Sign-in user using MSAL and obtain an access token for MS Graph
-                GraphServiceClient graphClient = new GraphServiceClient(s_graphURL,
-                    new DelegateAuthenticationProvider(async (requestMessage) =>
+                async Task<AuthenticationHeaderValue> SignInUserAndGetTokenUsingMSAL()
+                {
+                    IEnumerable<IAccount> accounts = await s_publicClientApp.GetAccountsAsync().ConfigureAwait(false);
+                    IAccount firstAccount = accounts.FirstOrDefault();
+
+                    try
                     {
-                        IEnumerable<IAccount> accounts = await s_publicClientApp.GetAccountsAsync().ConfigureAwait(false);
-                        IAccount firstAccount = accounts.FirstOrDefault();
-
-                        try
-                        {
-                            // Signs in the user and obtains an Access token for MS Graph
-                            s_authResult = await s_publicClientApp.AcquireTokenSilent(s_scopes, firstAccount)
-                                                              .ExecuteAsync();
-                        }
-                        catch (MsalUiRequiredException ex)
-                        {
-                            // A MsalUiRequiredException happened on AcquireTokenSilentAsync. This indicates you need to call AcquireTokenAsync to acquire a token
-                            Debug.WriteLine($"MsalUiRequiredException: {ex.Message}");
-
-                            s_authResult = await s_publicClientApp.AcquireTokenInteractive(s_scopes)
-#if ANDROID
-                                                              .WithParentActivityOrWindow(Microsoft.Maui.Essentials.Platform.CurrentActivity)
-#endif
-                                                              .ExecuteAsync()
-                                                              .ConfigureAwait(false);
-                        }
+                        // Signs in the user and obtains an Access token for MS Graph
+                        s_authResult = await s_publicClientApp.AcquireTokenSilent(s_scopes, firstAccount)
+                                                            .ExecuteAsync();
+                    }
+                    catch (MsalUiRequiredException ex)
+                    {
+                        // A MsalUiRequiredException happened on AcquireTokenSilentAsync. This indicates you need to call AcquireTokenAsync to acquire a token
+                        Debug.WriteLine($"MsalUiRequiredException: {ex.Message}");
 
-                        requestMessage.Headers.Authorization = new AuthenticationHeaderValue("bearer", s_authResult.AccessToken);
-                    }));
+                        s_authResult = await s_publicClientApp.AcquireTokenInteractive(s_scopes)

```

Comment:
This stopped working on android because you removed the 
```
#if ANDROID
.WithParentActivityOrWindow(Microsoft.Maui.Essentials.Platform.CurrentActivity)
```

bit.  Can you please check your merge again and make sure that you are carrying this over properly?

Reply:
You can move this up to the `ublicClientApplicationBuilder.Create` line and just use the existing `#if ANDROID` check up there so you don't need it down here.   That'll save the content dev team from having to "fork" the conversation twice in here.

```csharp
#if ANDROID
                    .WithRedirectUri($"msal{s_clientId}://auth")
                    .WithParentActivityOrWindow(() => Platform.CurrentActivity)
#else
                    .WithRedirectUri($"https://login.microsoftonline.com/common/oauth2/nativeclient")
#endif
```